### PR TITLE
Add 3.0.0 osd test manifest

### PR DESCRIPTION
**Pull request by Divyaasm**
### Description
This PR adds OSD componets for 3.0.0 test manifest

### Issues Resolved

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by gaiksaya — APPROVED** [review]

File: manifests/3.0.0/opensearch-dashboards-3.0.0-test.yml [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+  - name: queryWorkbenchDashboards
+    integ-test:
+      test-configs:
+        - with-security
+        - without-security
+  - name: securityDashboards
+    integ-test:
+      test-configs:
+        - with-security
+        - without-security
```
Comment (gaiksaya) [inline manifests/3.0.0/opensearch-dashboards-3.0.0-test.yml:42]: I dont think `securityDashboards` will have tests without-security. Can you confirm

Comment (gaiksaya) [reply manifests/3.0.0/opensearch-dashboards-3.0.0-test.yml:42]: @kavilla @tianleh Can you guys take a look and suggest changes to this?
Thanks!

Comment (gaiksaya) [reply manifests/3.0.0/opensearch-dashboards-3.0.0-test.yml:42]: Tagging security dashboards maintainers: @cwperks @peternied @davidlago

Comment (peternied) [reply manifests/3.0.0/opensearch-dashboards-3.0.0-test.yml:42]: I'm not sure how these tests are being sourced, this file looks different from 2.7.0 version, https://github.com/opensearch-project/opensearch-build/blob/main/manifests/2.7.0/opensearch-dashboards-2.7.0-test.yml , has there been changes to the architecture?

There are security tests in the `functionalTestDashboards` repo, which allows for running with and without security enabled.

Comment (gaiksaya) [reply manifests/3.0.0/opensearch-dashboards-3.0.0-test.yml:42]: Yes the architecture is changing. Thanks for asking this. Looks like we need to add documentation for new architecture! @Divyaasm Can you take this question up and also later take care of adding the documentation.

Comment (Divyaasm) [reply manifests/3.0.0/opensearch-dashboards-3.0.0-test.yml:42]: Yes @peternied , we have recently updated the implemention for OSD integtests, this used run in a sequential order for all the OSD component plugins listed in ftrepo based on functionalTestDashboards input earlier. Now OSD integtest works similar to OS intgtests. Refer the related PR's https://github.com/opensearch-project/opensearch-build/pull/3316 and https://github.com/opensearch-project/opensearch-build/pull/3465 to get an idea on the implementation changes.

We will now be able to to run this individually for each OSD component or all together at once in parallel from Jenkins perspective and update results individually for each component in the S3 bucket based on the updated test manifest similar to the current PR. Let me know if you have any questions.

This PR is just used to test the latest implementation from Prod Jenkins.

Thanks,
Divya Madala

Comment (peternied) [reply manifests/3.0.0/opensearch-dashboards-3.0.0-test.yml:42]: OK, well please go ahead and include the tests as has been done with other plugins.  I am not certain this is correct, but I think its best to move forward with generation results than waiting on making a determination

**Review by gaiksaya — APPROVED** [review]

File: manifests/3.0.0/opensearch-dashboards-3.0.0-test.yml (hunk 1 above)
Comment (gaiksaya) [reply manifests/3.0.0/opensearch-dashboards-3.0.0-test.yml:42]: Thanks for the input!

**Pull request merged**